helix-term/src/ui/editor.rs

```diff
@@ -476,6 +482,15 @@ impl EditorView {
             Mode::Select => "SEL",
             Mode::Normal => "NOR",
         };
+        let progress = doc
+            .language_server()
+            .and_then(|srv| {
+                spinners_map
```

Comment:
Let's keep the spinners map as spinners on ui::Editor, not on the Application type?

Reply:
`Editor` is still not accesible from this function, I'd have to pass it all the way from implementation of `Component` for `EditorView` so there would be no point in passing the other parameters. I'm fine with refactoring this if that is what you meant?

Reply:
Yeah sorry, I meant put the spinners into ui::EditorView since they're a part of display. You can get access to the editorview inside Application via  https://github.com/helix-editor/helix/blob/fc39a6c40df7d4a28994db762554815c37470073/helix-term/src/commands.rs#L2768-L2773

It's guaranteed to always be present so the unwrap is safe

Reply:
Done, its much cleaner with the spinners on the EditorView.